packages/conventional-changelog/index.js

```diff
@@ -6,7 +6,16 @@ function conventionalChangelog(options, context, gitRawCommitsOpts, parserOpts,
 
   if (options.preset) {
     try {
-      options.config = require('conventional-changelog-' + options.preset.toLowerCase());
+      var scope = '';
+      var name = options.preset.toLowerCase();
+
+      if (name[0] === '@') {
+        var parts = name.split('/');
+        scope = parts.shift() + '/';
+        name = parts.join('');
```

Comment:
There will never be anything here to join. A forward slash may only be used to delineate between a scope and package name. [Here's the relevant validation code](https://github.com/npm/validate-npm-package-name/blob/393f65720d07eca49e9f1672f0cf001b71b021d9/index.js#L77).

Reply:
`npm` may validate package names like this, but I'm not sure that this validation is done in `conventional-changelog`. The package is simply required, so if someone was previously passing `--preset my-preset/lib/file.js` that would translate to `require('conventional-changelog-my-preset/lib/file.js')` which is valid.

I think this is needed to avoid accidental breakage

Reply:
Thank you for catching that.

Reply:
@evocateur what about the following scenario:

You list the preset as `@mycompany/angular` with the expectation that it will load the `@mycompany/conventional-changelog-angular` package listed in your development dependencies. One of your dependencies lists `@mycompany/angular` as a dependency, which is installed at the top level `node_modules/` directory because `npm` and `yarn` both flatten the `node_modules/` directory as best they can.

When `conventional-changelog` goes to `require` the `preset` you listed, which is `@mycompany/angular`, it would try loading the actual `@mycompany/angular` package, not the `@mycompany/conventional-changelog-angular` package you intended.